### PR TITLE
Make clang-format bind pointers to the type (left)

### LINKED ISSUE #297: Clang Format Inconsistencies
Currently we don't prescribe how pointers should be handled in the clang format file, and instead leave it to be inferred (via `DerivePointerAlignment`). This means: 

1. We have inconsistent formatting in the code; and 
2. It allows different versions of clang format to do different things (which is currently causing me issues, where my format disagrees with Travis). 

I propose we switch to specifying a value for `PointerAlignment`, and will make a PR. It looks like in the code currently we (mostly) do middle alignment for pointers and right alignment for references...

### PR DESCRIPTION
**Pull request by rfbird**
Previously pointer alignment was "derived" based on context in the code.
This PR explicitly specifies it should be left binding, which should
give more consistent behaviour across clang format versions

The changes in this PR are cosmetic only.

Fixes #297

There is still a difference between how clang 9 and clang 10 format code. We can make 10 act like 9 by specifying `Cpp11BracedListStyle = false`, but I can't find a way to make 9 act like 10....

I'll try up the minimum version in another PR

### CODE REVIEW

File: cajita/src/Cajita_Halo.hpp [hunk 1]
```diff
@@ -519,11 +519,11 @@ class Halo
     // alignment boundaries.
     template <class ArrayView>
     KOKKOS_INLINE_FUNCTION void
-    packElement( const Kokkos::View<char *, memory_space> &buffer,
-                 const Kokkos::View<int *[6], memory_space> &steering,
-                 const int element_idx, const ArrayView &array_view ) const
+    packElement( const Kokkos::View<char*, memory_space>& buffer,
+                 const Kokkos::View<int* [6], memory_space>& steering,
```
Comment (sslattery) [inline cajita/src/Cajita_Halo.hpp:523]: Too bad these view data types get spaces in them.

**Review by sslattery — APPROVED** [review]

File: cajita/src/Cajita_Halo.hpp (hunk 1 above)
Comment (rfbird) [reply cajita/src/Cajita_Halo.hpp:523]: What would you prefer to see? I can try twiddle the options

Comment (sslattery) [reply cajita/src/Cajita_Halo.hpp:523]: it would be nice if there was no space in Kokkos view data types but that might be hard.

Comment (rfbird) [reply cajita/src/Cajita_Halo.hpp:523]: So ` Kokkos::View<int*[6],memory_space>&` but only for kokkos views? Yea, that sounds hard..

**Comment by streeve** [comment]
Is this ready to merge?

**Comment by rfbird** [comment]
> Is this ready to merge?

Yes from my perspective

**Pull request merged**